scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -464,7 +469,23 @@ def __init__(self, max_failure_count=5, blacklist_time_secs=300, max_query_retri
         self.__last_reported_warming_attempts = dict()
 
     def start(self):
-        super(ControlledCacheWarmer, self).start()
+        self.__lock.acquire()
+        try:
+            super(ControlledCacheWarmer, self).start()
+            self.__is_running = True
```

Comment:
nit: 

Wondering can we just use `StoppableThread._run_state.is_running` instead?

Maybe expose a is_running property on `StoppableThread`

This should avoid duplicating this state here

Reply:
I'm also suspect this new non-deterministic exception occuring in CI and locally may be caused by a race condition because we use a separate is_running flag (instead of relying on StoppableThread runstate's flag.
 
```
======================================================================
ERROR: test_already_warm (scalyr_agent.builtin_monitors.tests.kubernetes_monitor_test.ControlledCacheWarmerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/scalyr-agent-2/scalyr_agent/builtin_monitors/tests/kubernetes_monitor_test.py", line 209, in test_already_warm
    warmer.block_until_idle( self.timeout )
  File "/home/circleci/scalyr-agent-2/scalyr_agent/builtin_monitors/kubernetes_monitor.py", line 706, in block_until_idle
    raise Exception( "block_until_idle timed out" )
Exception: block_until_idle timed out
```

Reply:
Yup, it's a good suggestion to re-use run_state.  I should have thought of that.  I'll take a look at the non-deterministic failure.  I'll have a revision in the next hour for this PR.

Reply:
After digging into this more (and discussing with you), since run_state does not know when the thread is actually started, we are not going to re-use it.  It would require too risky of a refactor to change run_state to track start times.